From 0116ff4f920173ed1ad51699d12e2084f3f16d05 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sat, 22 May 2021 18:38:13 -0500
Subject: [PATCH 3/3] deps: link libtiff first

libtiff sometimes depends on libjpeg etc and linking it last
will cause linking issues even if the others are loaded manually

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 CMakeLists.txt   |  2 +-
 cmake/deps.cmake | 16 +++-------------
 2 files changed, 4 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f51fdd9..af10f1bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -422,7 +422,7 @@ if(WEBP_BUILD_ANIM_UTILS
                         imageioutil
                         webpdemux
                         webp
-                        ${WEBP_DEP_IMG_LIBRARIES})
+                        ${WEBP_DEP_IMG_LDFLAGS})
 
   # Image-encoding utility library.
   parse_makefile_am(${CMAKE_CURRENT_SOURCE_DIR}/imageio "IMAGEENC_SRCS"
diff --git a/cmake/deps.cmake b/cmake/deps.cmake
index e00611f6..a77793ed 100644
--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -71,19 +71,9 @@ if(NOT HAVE_MATH_LIBRARY)
 endif()
 
 # Find the standard image libraries.
-set(WEBP_DEP_IMG_LIBRARIES)
-set(WEBP_DEP_IMG_INCLUDE_DIRS)
-foreach(I_LIB PNG JPEG TIFF)
-  find_package(${I_LIB})
-  set(WEBP_HAVE_${I_LIB} ${${I_LIB}_FOUND})
-  if(${I_LIB}_FOUND)
-    list(APPEND WEBP_DEP_IMG_LIBRARIES ${${I_LIB}_LIBRARIES})
-    list(APPEND WEBP_DEP_IMG_INCLUDE_DIRS ${${I_LIB}_INCLUDE_DIR}
-                ${${I_LIB}_INCLUDE_DIRS})
-  endif()
-endforeach()
-if(WEBP_DEP_IMG_INCLUDE_DIRS)
-  list(REMOVE_DUPLICATES WEBP_DEP_IMG_INCLUDE_DIRS)
+find_package(PkgConfig)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(WEBP_DEP_IMG libtiff-4 libjpeg libpng)
 endif()
 
 # GIF detection, gifdec isn't part of the imageio lib.
-- 
2.31.1

